x-pack/functionbeat/function/core/coordinator_test.go

```diff
@@ -36,21 +38,21 @@ func TestStart(t *testing.T) {
 		ctx, cancel := context.WithCancel(context.Background())
 		var err error
 		go func() {
-			err = coordinator.Run(ctx)
+			err = coordinator.Run(ctx, nil)
 			assert.NoError(t, err)
 		}()
 		cancel()
 	})
 
 	t.Run("on error shutdown all the runner", func(t *testing.T) {
 		coordinator := NewCoordinator(nil, &happyRunner{}, &unhappyRunner{})
-		err := coordinator.Run(context.Background())
+		err := coordinator.Run(context.Background(), nil)
```

Comment:
Interesting approach with the telemetry package. So far I like this idea better than having to remember that the 'stats' registry is for telemetry :)

Instead of passing nil, how about providing a function like `Ignore() T` in the `telemetry` package, that returns a `telemetry.T` instance doing nothing. We would use `telemetry.Ignored()` instead of passing a `nil` then.